webapp/src/components/root/root.jsx

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {makeStyleFromTheme} from 'mattermost-redux/utils/theme_utils';
+
+import {id as pluginId} from '../../manifest';
+
+import mmLogoURL from './../../images/mm-logo.png';
+import webexLogoURL from './../../images/webex-logo.png';
+
+const {Modal} = window.ReactBootstrap;
+
+export default class Root extends React.PureComponent {
+    static propTypes = {
+
+        theme: PropTypes.object.isRequired,
+        siteUrl: PropTypes.string.isRequired,
+        visible: PropTypes.bool.isRequired,
+        authenticated: PropTypes.object.isRequired,
+        actions: PropTypes.shape({
+            closeRootModal: PropTypes.func.isRequired,
+        }).isRequired,
+
+    }
+
+    handleClose = () => {
+        this.props.actions.closeRootModal();
+    };
+
+    startOAuthConnectFlow = async () => {
+        window.open(`${this.props.siteUrl}/plugins/${pluginId}/oauth2/connect`);
+    }
+
+    render() {
+        var posWidth = ((window.innerWidth - 200) + 'px');
+        var style = getStyle(posWidth, this.props.theme);
+
+        return (
+            <div style={style.modelCont}>
+
+                <Modal
+                    show={this.props.visible}
+                    onHide={this.handleClose}
+                    style={style.modal}
+                >
+                    <Modal.Header
+                        closeButton={true}
+                        style={style.header}
+                    ></Modal.Header>
```

Comment:
This should be just a self-closed tag.
```suggestion
                   <Modal.Header
                        closeButton={true}
                        style={style.header}
                    />
```